### PR TITLE
Typescript support

### LINKED ISSUE #14: support typescript
support typescript

### PR DESCRIPTION
**Pull request by chyzwar**
Provide typescript support. Based on https://github.com/fastify/fastify-oauth2/pull/35.
resolves #14

Facebook Graph API version upgraded from 3 to 6.  I tested and review changelog
https://developers.facebook.com/docs/graph-api/changelog/version6.0

Additionally, I upgraded dev dependencies.

### CODE REVIEW

**Comment by chyzwar** [comment]
@Eomm 
CI is failing on node 6. Is fastify still supporting node 6 and 8 ?

**Comment by allevo** [comment]
Nodejs 6 and 8 is still supported according with 
https://github.com/fastify/fastify/blob/a6db28188a8ea4fcc66cbbd4caee2ef126687525/.github/workflows/ci.yml#L12-L20

**Comment by allevo** [comment]
But tap doesn't:
https://github.com/tapjs/node-tap/blob/1ef056f28db6dce710a24ff33c2519defe1a34f7/package.json#L12

If you revert the "tap" upgrade keeping 12, the tests should pass

File: package.json [hunk 1]
```diff
@@ -25,12 +25,12 @@
   "homepage": "https://github.com/fastify/fastify-oauth2#readme",
   "devDependencies": {
     "fastify": "^2.0.1",
-    "nock": "^9.3.2",
+    "nock": "^12.0.3",
```
Comment (mcollina) [inline package.json]: Please restore this to v9, thanks.

Comment (chyzwar) [reply package.json]: I reverted to v9. 

Fastify plan to drop node 6 and 8?  Currently, this plugin uses simple-oauth2 v1  that is no longer supported. Upgrading to v3 would be possible but require dropping v6.

Comment (Eomm) [reply package.json]: > Fastify plan to drop node 6 and 8?

Yes, we will release Fastify v3 soon.
Check out the main repo for more info

**Pull request merged**